### PR TITLE
Initialize analysis incubator repository

### PR DESCRIPTION
**Pull request by navarroc**
This PR creates the basic incubator repository for new analyses not ready to be included in the core set of analyses. Included are:
- [x] boilerplate for new pyincore-incubator including license, setup.py, readme.rst, tests, .gitignore, etc
- [x] docs folder for module documentation
- [x] example analysis and test
- [x] python formatting checks (black and flake8)
- [x] github workflows for conda, pypi and doc build

### CODE REVIEW

File: tests/pyincore-incubator/analyses/example/test_example.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from pyincore import IncoreClient
```
Comment (longshuicy) [inline tests/pyincore-incubator/analyses/example/test_example.py:1]: Does this folder need to match the analysis folder and to be changed to use underscore "pyincore_incubator"?

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
+pyincore
```
Comment (longshuicy) [inline README.rst:1]: Title is pyincore? is that intended?

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+========
+
+**pyIncore-incubator** is a component of IN-CORE. It is a python package
+consisting of new analyses that build on pyIncore, but are not yet part of the
+core analyses.
+
+Installation with conda
+-----------------------
+
+Installing **pyincore** with Conda is officially supported by IN-CORE development team.
```
Comment (longshuicy) [inline README.rst:11]: Same here, the instruction seems to be about installing pyincore-incubator but here it says "pyincore". 
Also is the dev team gonna support "pyincore-incubator"?

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,77 @@
+pyincore
+========
+
+**pyIncore-incubator** is a component of IN-CORE. It is a python package
+consisting of new analyses that build on pyIncore, but are not yet part of the
+core analyses.
```
Comment (longshuicy) [inline README.rst:6]: maybe add a sentence to explain its connection with pyincore?

**Review by longshuicy — CHANGES_REQUESTED** [review]
I took a first pass review. Tested pre-commit, sphinx, and test analysis and it all works. Left you some minor comments.

Two more things:
1. Do we want to replace the logo to use the new design?
2. Could we add some text instructing user to set up pre-commit? 

Here is what we wrote in clowder doc

----

We use [pre-commit](https://pre-commit.com/) to run linting and formatting checks before committing code. To install pre-commit, run `pip install pre-commit` and then `pre-commit install` in the root directory of the repository. This will install the pre-commit hooks and run them on every commit. If any of the hooks fail, the commit will be aborted. 
Once ran, the hooks will be in place for all future commits and branches since they are stored in the .git/hooks. To run the hooks manually, run `pre-commit run --all-files`. To skip the hooks, run `git commit --no-verify` (or click on cog in PyCharm and uncheck Run Git hooks).

See .pre-commit-config.yaml for the list of hooks that are run. The hooks are run in the order they are listed in the file. The hooks are run on all files that are staged for commit. To run the hooks on all files, run `pre-commit run --all-files`.
Hooks include:
* Python style checks and formatting using black and flake8.

File: tests/pyincore-incubator/analyses/example/test_example.py (hunk 1 above)
Comment (navarroc) [reply tests/pyincore-incubator/analyses/example/test_example.py:1]: Good catch, let me fix that.

File: README.rst (hunk 2 above)
Comment (navarroc) [reply README.rst:1]: Another good catch, will fix this.

File: README.rst (hunk 3 above)
Comment (navarroc) [reply README.rst:11]: That's a good question. I talked with Jong a little and I think the idea would be that we will support the conda installation and maintaining that it works and publishes packages on some release schedule, but contributors would be responsible for making sure their code still works with new releases.

File: README.rst (hunk 4 above)
Comment (navarroc) [reply README.rst:6]: Something like this? If not, can you suggest what's missing and I can try to rework the text? Thanks!

**pyIncore-incubator** is a component of IN-CORE. It is a python package
consisting of new analyses that build on pyIncore, but are not yet part of the 
core analyses. Analyses extend the BaseAnalysis class from pyIncore and add new
functionality.

**Comment by navarroc** [comment]
> I took a first pass review. Tested pre-commit, sphinx, and test analysis and it all works. Left you some minor comments.
> 
> Two more things:
> 
> 1. Do we want to replace the logo to use the new design?
> 2. Could we add some text instructing user to set up pre-commit?
> 
> Here is what we wrote in clowder doc
> 
> We use [pre-commit](https://pre-commit.com/) to run linting and formatting checks before committing code. To install pre-commit, run `pip install pre-commit` and then `pre-commit install` in the root directory of the repository. This will install the pre-commit hooks and run them on every commit. If any of the hooks fail, the commit will be aborted. Once ran, the hooks will be in place for all future commits and branches since they are stored in the .git/hooks. To run the hooks manually, run `pre-commit run --all-files`. To skip the hooks, run `git commit --no-verify` (or click on cog in PyCharm and uncheck Run Git hooks).
> 
> See .pre-commit-config.yaml for the list of hooks that are run. The hooks are run in the order they are listed in the file. The hooks are run on all files that are staged for commit. To run the hooks on all files, run `pre-commit run --all-files`. Hooks include:
> 
> * Python style checks and formatting using black and flake8.

Yes, we should probably use the new logo. For your other point, it's not clear where that should go. The README.rst is more about installing the package for running, not for development. The precommit hook is more about development and that should probably go wherever we put the developer documentation. Maybe a wiki page in IN-CORE docs?

**Comment by navarroc** [comment]
I will update the logo since we are changing it for the project, but it's not actually used anywhere - FYI

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
+## pyIncore_incubator documentation
+
+**pyIncore_incubator** documentation which is built using Python [Sphinx](http://www.sphinx-doc.org/en/master/) package.
+
+### Installation
+
+Clone the code from pyincore repository [git](https://github.com/IN-CORE/pyincore-incubator.git)
```
Comment (longshuicy) [inline docs/README.md:7]: Clone the code from `pyincore-incubator`?

File: recipes/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+source:
+  path: ..
+
+about:
+  home: https://incore.ncsa.illinois.edu
+  license: MPL-2.0
+  summary: 'Python library for IN-CORE (Interdependent Networked Community Resilience Modeling Environment)'
+  description: 'pyIncore is a component of IN-CORE. It is a python package consisting of two primary components: 
+  1) a set of service classes to interact with the IN-CORE web services, and 2) IN-CORE analyses. The pyIncore
```
Comment (longshuicy) [inline recipes/meta.yaml:16]: Do we need to adjust the description here, or is it intended to describe pyincore?

File: requirements.sh [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+# ${MODULE} dependencies version list
+
+This file is autogenerated and lists the latest versions of all dependencies. The table will list package (pip and conda), minimum version and latest version found.
+
+| PIP Package | Conda Package | Required Version | Matched Version (pip) |
+|-------------|---------------|------------------|-----------------------|
+EOF
+
+# create the environment.yml file for conda. This is intended to setup a conda environment for
+# development on pyincore_viz.
```
Comment (longshuicy) [inline requirements.sh:77]: pyincore_viz?

File: setup.py [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    extras_require={
+        "test": [
+            "pycodestyle>=2.6.0",
+            "pytest>=3.9.0",
+            "python-jose>=3.0",
+        ]
+    },
+    project_urls={
+        "Bug Reports": "https://github.com/IN-CORE/pyincore/issues",
+        "Source": "https://github.com/IN-CORE/pyincore",
```
Comment (longshuicy) [inline setup.py:60]: project_urls points to pyincore still

**Review by longshuicy — APPROVED** [review]
I did a project wide search of "pyincore" and find a few more missing spots. Besides that all seems to look good.

File: requirements.sh (hunk 7 above)
Comment (navarroc) [reply requirements.sh:77]: another good find.

File: docs/README.md (hunk 5 above)
Comment (navarroc) [reply docs/README.md]: fixed

File: recipes/meta.yaml (hunk 6 above)
Comment (navarroc) [reply recipes/meta.yaml]: Fixed this too

**Comment by ylyangtw** [comment]
Tested docker container and local env for sphinx, also tested example analysis. All work.
For my machine, I still had to specify `--platform linux/x86_64` to build the container.

File: README.rst [hunk 9]
```diff
@@ -0,0 +1,83 @@
+pyincore-incubator
+========
+
```
Comment (ylyangtw) [inline README.rst:3]: All look good. Maybe we don't need this line? `pyincore-incubator ========`
image

**Review by ylyangtw — APPROVED** [review]

**Pull request merged**